### PR TITLE
[istio] Fixed operability in managed k8s setups

### PR DESCRIPTION
**Pull request by Beastlex**
## Description
Fixed istio module operability in managed k8s setups.

## Why do we need it, and what problem does it solve?
There is no way to run the istio module and its tests on managed clusters with minimal bundle (such EKS).

## Why do we need it in the patch release (if we do)?
There is a customer who considers using istio in EKS.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

```changes
section: istio
type: fix
summary: Fixed istio module operability in managed k8s setups.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/110-istio/hooks/discovery_preflight_check.go [hunk 1]
```diff
@@ -78,17 +79,24 @@ func applyClusterConfigurationYamlFilter(obj *unstructured.Unstructured) (go_hoo
 }
 
 func discoveryIsK8sVersionAutomatic(input *go_hook.HookInput) error {
+	var kubernetesVersionStr string
 	kubernetesVersion, ok := input.Snapshots["kubernetesVersion"]
```
Comment (apolovov) [inline modules/110-istio/hooks/discovery_preflight_check.go:83]:
```suggestion
	clusterConfigurationSnapshots, ok := input.Snapshots["kubernetesVersion"]
```

File: modules/110-istio/hooks/discovery_preflight_check.go [hunk 2]
```diff
@@ -78,17 +79,24 @@ func applyClusterConfigurationYamlFilter(obj *unstructured.Unstructured) (go_hoo
 }
 
 func discoveryIsK8sVersionAutomatic(input *go_hook.HookInput) error {
-	kubernetesVersion, ok := input.Snapshots["kubernetesVersion"]
-	if !ok || len(kubernetesVersion) == 0 {
-		return errors.New("cluster configuration kubernetesVersion is empty or invalid")
+	var kubernetesVersionStr string
+	clusterConfigurationSnapshots, ok := input.Snapshots["kubernetesVersion"]
```
Comment (apolovov) [inline modules/110-istio/hooks/discovery_preflight_check.go:83]:
```suggestion
	clusterConfigurationSnapshots, ok := input.Snapshots["cluster-configuration"]
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/eks'...

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9326](https://github.com/deckhouse/deckhouse/pull/9326) to the branch [release-1.63](https://github.com/deckhouse/deckhouse/tree/release-1.63) successful!

**Pull request merged**